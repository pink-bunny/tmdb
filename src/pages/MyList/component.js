import React from 'react';
import PropTypes from 'prop-types';
import {
  Layout, Row, Col, Typography, Icon
} from 'antd';
import List from '../../components/List';
import DeleteListModal from '../../components/DeleteListModal';

const MyList = ({
  listInfo,
  listMovies,
  loading,
  error,
  totalItems,
  currentPage,
  fetchList,
  removeModalAction
}) => (
  <Layout>
    <Layout.Content>
      <Row>
        <Col
          offset={2}
          span={20}
        >
          <div className="top-margin">
            <Typography.Title>
              {listInfo && (
                <>
                  {listInfo.name}
                  {' '}
                  <DeleteListModal
                    id={listInfo.id}
                    name={listInfo.name}
                    key="delete"
                    triggerComponent={Icon}
                    triggerProps={{ type: 'minus-circle' }}
                  />
                </>
              )}
            </Typography.Title>
          </div>
        </Col>
      </Row>

      <List
        emptyListTxt={`No movies in ${listInfo && listInfo.name} found`}
        list={listMovies}
        loading={loading}
        error={error}
        totalItems={totalItems}
        currentPage={currentPage}
        fetchList={fetchList}
        removeModalAction={removeModalAction}
        removeModalTxt={`Do you want to delete this item from the ${listInfo && listInfo.name}?`}
      />

    </Layout.Content>
  </Layout>
);

MyList.propTypes = {
  fetchList: PropTypes.func.isRequired,
  removeModalAction: PropTypes.func.isRequired,
  loading: PropTypes.bool.isRequired,
  listInfo: PropTypes.objectOf(PropTypes.any),
  error: PropTypes.string.isRequired,
  totalItems: PropTypes.number.isRequired,
  currentPage: PropTypes.number.isRequired,
  listMovies: PropTypes.arrayOf(PropTypes.object)
};
MyList.defaultProps = {
  listInfo: null,
  listMovies: null
};

export default MyList;
