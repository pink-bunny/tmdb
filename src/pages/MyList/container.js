import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import List from '../../components/List';
import {
  fetchMyList as fetchMyListAction,
  removeFromList as removeFromListAction
} from '../../../state/my_list/actions';
import {
  isMyListLoading,
  myListError,
  myListInfo
} from '../../../state/my_list/selectors';

class MyListContainer extends React.Component {
  handleFetchList = () => {
    const { match, fetchMyList } = this.props;
    const { listId } = match.params;

    fetchMyList(listId);
  }

  handleRemoveFromList = (movieId) => {
    const { match, removeFromList } = this.props;
    const { listId } = match.params;

    removeFromList(listId, movieId);
  }

  render() {
    const {
      listInfo,
      loading,
      error,
      totalItems,
      currentPage
    } = this.props;
    return (
      <List
        listTitle={listInfo ? listInfo.name : ''}
        emptyListTxt={`No movies in ${listInfo && listInfo.name} found`}
        list={listInfo && listInfo.items}
        loading={loading}
        error={error}
        totalItems={totalItems}
        currentPage={currentPage}
        fetchList={this.handleFetchList}
        removeModalAction={this.handleRemoveFromList}
        removeModalTxt={`Do you want to delete this item from the ${listInfo && listInfo.name}?`}
      />
    );
  }
}

MyListContainer.propTypes = {
  fetchMyList: PropTypes.func.isRequired,
  removeFromList: PropTypes.func.isRequired,
  loading: PropTypes.bool.isRequired,
  listInfo: PropTypes.objectOf(PropTypes.any),
  error: PropTypes.string.isRequired,
  totalItems: PropTypes.number.isRequired,
  currentPage: PropTypes.number.isRequired,
  match: PropTypes.shape({
    params: PropTypes.shape({
      listId: PropTypes.string.isRequired
    })
  }).isRequired
};
MyListContainer.defaultProps = {
  listInfo: null
};

const mapDispatchToPtops = {
  fetchMyList: fetchMyListAction,
  removeFromList: removeFromListAction
};

const mapStateToProps = (state) => ({
  loading: isMyListLoading(state),
  listInfo: myListInfo(state),
  error: myListError(state),
  totalItems: 0,
  currentPage: 0
});

export default connect(mapStateToProps, mapDispatchToPtops)(MyListContainer);
