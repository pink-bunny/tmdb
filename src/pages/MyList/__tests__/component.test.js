import React from 'react';
import { shallow } from 'enzyme';

import MyList from '../component';

describe('MyList component', () => {
  const requiredProps = {
    loading: false,
    error: '',
    totalItems: 21,
    currentPage: 2,
    fetchList: jest.fn(),
    removeModalAction: jest.fn()
  };
  const wrapper = shallow(<MyList {...requiredProps} />); /* eslint-disable-line react/jsx-props-no-spreading, max-len */

  it('matches snapshot with required props', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('matches snapshot with optional props', () => {
    wrapper.setProps({
      listInfo: { id: 1, name: 'Custom list' },
      listMovies: [{ id: 1, title: 'Mock title' }]
    });
    expect(wrapper).toMatchSnapshot();
  });
});
