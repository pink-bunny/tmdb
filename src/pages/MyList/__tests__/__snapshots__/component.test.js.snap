// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyList component matches snapshot with optional props 1`] = `
<Adapter>
  <Adapter>
    <Row
      gutter={0}
    >
      <Col
        offset={2}
        span={20}
      >
        <div
          className="top-margin"
        >
          <Title>
            Custom list
             
            <Connect(WrappedComponentShell)
              id={1}
              key="delete"
              name="Custom list"
              needRedirect={true}
              triggerComponent={[Function]}
              triggerProps={
                Object {
                  "type": "minus-circle",
                }
              }
            />
          </Title>
        </div>
      </Col>
    </Row>
    <ListContainer
      currentPage={2}
      emptyListTxt="No movies in Custom list found"
      error=""
      fetchList={[MockFunction]}
      list={
        Array [
          Object {
            "id": 1,
            "title": "Mock title",
          },
        ]
      }
      loading={false}
      removeModalAction={[MockFunction]}
      removeModalTxt="Do you want to delete this item from the Custom list?"
      totalItems={21}
    />
  </Adapter>
</Adapter>
`;

exports[`MyList component matches snapshot with required props 1`] = `
<Adapter>
  <Adapter>
    <Row
      gutter={0}
    >
      <Col
        offset={2}
        span={20}
      >
        <div
          className="top-margin"
        >
          <Title>
            List
             
          </Title>
        </div>
      </Col>
    </Row>
    <ListContainer
      currentPage={2}
      emptyListTxt="No movies in this list found"
      error=""
      fetchList={[MockFunction]}
      list={null}
      loading={false}
      removeModalAction={[MockFunction]}
      removeModalTxt="Do you want to delete this item from the list?"
      totalItems={21}
    />
  </Adapter>
</Adapter>
`;
